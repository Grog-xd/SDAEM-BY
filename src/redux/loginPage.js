import {createSlice} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, setBookMarkActive, registerUser} = loginPage.actions

