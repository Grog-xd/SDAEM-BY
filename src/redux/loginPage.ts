import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'
import {profilesArr} from '../types/types';

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action:PayloadAction<profilesArr>){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action:PayloadAction<any>){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, registerUser} = loginPage.actions

