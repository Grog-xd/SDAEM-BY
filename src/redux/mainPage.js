import {createSlice} from '@reduxjs/toolkit';

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:'',
        maxCost:'',


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action){
            state.sortedValue = action.payload
        },

        setLimit(state, action){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = ''
            state.maxCost = ''
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = +state.maxCost + 100
            }
            if(state.minCost === ''){
                state.minCost=0
            }
            if(state.maxCost === ''){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
