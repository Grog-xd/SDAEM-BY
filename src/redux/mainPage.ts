import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import {ProductProps} from '../types/types';


interface likeActionProps{
    id: number,
    type:string
}

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:0,
        maxCost:1000,


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action:PayloadAction<string>){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action:PayloadAction<string>){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action:PayloadAction<likeActionProps>){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action:PayloadAction<string>){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action:PayloadAction<string>){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action:PayloadAction<number>){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action:PayloadAction<number>){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action:PayloadAction<string>){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action:PayloadAction<string>){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action:PayloadAction<number>){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action:PayloadAction<string>){
            state.sortedValue = action.payload
        },

        setLimit(state, action:PayloadAction<number>){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = undefined
            state.maxCost = undefined
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action:PayloadAction<ProductProps>){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = state.maxCost + 100
            }
            if(state.minCost === undefined){
                state.minCost = 0
            }
            if(state.maxCost === undefined){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes?.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes?.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
