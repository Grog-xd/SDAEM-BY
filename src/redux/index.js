import {combineReducers} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import mainPage from "./mainPage";
import newsPage from "./newsPage";
import loginPage from "./loginPage";

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage
})

export const store = configureStore({
    reducer: rootReducer
})
