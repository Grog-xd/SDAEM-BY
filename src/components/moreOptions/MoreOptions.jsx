import React from 'react';
import {useDispatch, useSelector} from "react-redux";
import {useParams} from "react-router-dom";
import {checkboxHandler, setBedSum, setDistrict, setMetro} from "../../redux/mainPage";
import MySelect from "../UI/mySelect/mySelect";
import CheckboxList from "./checkboxList/CheckboxList";
import classes from './MoreOptions.module.scss'


const MoreOptions = () => {
    const dispatch = useDispatch()
    const params = useParams()
    const {bedSumOptions, bedSumCurrentValue, metroOption, metroCurrentValue, districtOption, districtCurrentValue, checkboxsMoreOption} = useSelector(state=> state.main)

    return (
        <div className={classes.moreOptions}>
            <div className={classes.selectSection}>
                {
                    params.type !== 'cars'
                        &&
                            <div className={classes.selectBlock}>
                                <p className={classes.selectLabel}>Спальные места</p>
                                <MySelect options={bedSumOptions} handler={(e)=> dispatch(setBedSum(e))} value={bedSumCurrentValue ? bedSumCurrentValue: 'Выберите'} style={classes.select}/>
                            </div>
                }

                <div className={classes.selectBlock}>
                    <p className={classes.selectLabel}>Район</p>
                    <MySelect options={districtOption} handler={(e)=> dispatch(setDistrict(e))} value={districtCurrentValue ? districtCurrentValue: 'Выберите'} style={classes.select}/>
                </div>
                <div className={classes.selectBlock}>
                    <p className={classes.selectLabel}>Метро</p>
                    <MySelect options={metroOption} handler={(e)=> dispatch(setMetro(e))} value={metroCurrentValue ? metroCurrentValue: 'Выберите'} style={classes.select}/>
                </div>
            </div>
            {
                params.type !== 'cars'
                    &&  <CheckboxList checkboxList={checkboxsMoreOption} handler={(e)=>dispatch(checkboxHandler(e))}/>
            }

        </div>
    );
};

export default MoreOptions;
