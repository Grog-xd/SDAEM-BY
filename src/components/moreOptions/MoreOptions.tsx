import {FC} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {useParams} from 'react-router-dom';

import {checkboxHandler, setBedSum, setDistrict, setMetro} from '../../redux/mainPage';
import MySelect from '../UI/mySelect/mySelect.tsx';

import {mainRedux} from '../../types/types';

import CheckboxList from './checkboxList/CheckboxList.tsx';
import classes from './MoreOptions.module.scss'


const MoreOptions:FC = () => {
    const dispatch = useDispatch()
    const params = useParams()
    const {bedSumOptions, bedSumCurrentValue, metroOption, metroCurrentValue, districtOption, districtCurrentValue, checkboxsMoreOption} = useSelector((state:mainRedux)=> state.main)

    return (
        <div data-testid={'more-options'} className={classes.moreOptions}>
            <div className={classes.selectSection}>
                {
                    params.type !== 'cars'
                        &&
                            <div className={classes.selectBlock}>
                                <p className={classes.selectLabel}>Спальные места</p>
                                <MySelect options={bedSumOptions} handler={(e)=> dispatch(setBedSum(e))} value={bedSumCurrentValue ? bedSumCurrentValue: 'Выберите'} style={classes.select}/>
                            </div>
                }

                <div className={classes.selectBlock}>
                    <p className={classes.selectLabel}>Район</p>
                    <MySelect options={districtOption} handler={(e)=> dispatch(setDistrict(e))} value={districtCurrentValue ? districtCurrentValue: 'Выберите'} style={classes.select}/>
                </div>
                <div className={classes.selectBlock}>
                    <p className={classes.selectLabel}>Метро</p>
                    <MySelect options={metroOption} handler={(e)=> dispatch(setMetro(e))} value={metroCurrentValue ? metroCurrentValue: 'Выберите'} style={classes.select}/>
                </div>
            </div>
            {
                params.type !== 'cars'
                    &&  <CheckboxList checkboxList={checkboxsMoreOption} handler={(e)=>dispatch(checkboxHandler(e))}/>
            }

        </div>
    );
};

export default MoreOptions;
