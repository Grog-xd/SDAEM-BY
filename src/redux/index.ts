import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {useDispatch} from 'react-redux';

import mainPage from './mainPage';
import newsPage from './newsPage';
import loginPage from './loginPage';

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage,
})

export const store = configureStore({
    reducer: rootReducer,
})


export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
