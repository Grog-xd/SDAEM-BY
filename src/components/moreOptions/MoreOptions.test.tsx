import {render, screen} from '@testing-library/react';
import {Provider} from 'react-redux';

import {store} from '../../redux';

import MoreOptions from './MoreOptions';

describe('more options tests', () =>{
    test('snapshot test', ()=>{
        render(
            <Provider store={store} >
                <MoreOptions />
            </Provider>,
        )
        const section = screen.getByTestId('more-options')
        expect(section).toMatchSnapshot();
    })
})
