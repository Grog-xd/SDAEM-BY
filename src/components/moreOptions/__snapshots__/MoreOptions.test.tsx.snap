// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`more options tests snapshot test 1`] = `
<div
  class="moreOptions"
  data-testid="more-options"
>
  <div
    class="selectSection"
  >
    <div
      class="selectBlock"
    >
      <p
        class="selectLabel"
      >
        Спальные места
      </p>
      <div
        class="select select"
      >
        <button
          class="changeBlock"
          type="button"
        >
          <div>
            <p
              class="textValue"
            >
              Выберите
            </p>
          </div>
          <svg
            fill="none"
            height="16"
            viewBox="0 0 12 7"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 1.38477L6 5.64152L10.5 1.38477"
              stroke="#4E64F9"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="selectBlock"
    >
      <p
        class="selectLabel"
      >
        Район
      </p>
      <div
        class="select select"
      >
        <button
          class="changeBlock"
          type="button"
        >
          <div>
            <p
              class="textValue"
            >
              Выберите
            </p>
          </div>
          <svg
            fill="none"
            height="16"
            viewBox="0 0 12 7"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 1.38477L6 5.64152L10.5 1.38477"
              stroke="#4E64F9"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="selectBlock"
    >
      <p
        class="selectLabel"
      >
        Метро
      </p>
      <div
        class="select select"
      >
        <button
          class="changeBlock"
          type="button"
        >
          <div>
            <p
              class="textValue"
            >
              Выберите
            </p>
          </div>
          <svg
            fill="none"
            height="16"
            viewBox="0 0 12 7"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 1.38477L6 5.64152L10.5 1.38477"
              stroke="#4E64F9"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class="checkboxSection"
  >
    <div
      class="checkbox"
    >
      <input
        id="Газовая плита"
        name="Газовая плита"
        type="checkbox"
      />
      <label
        for="Газовая плита"
      >
        Газовая плита
      </label>
    </div>
    <div
      class="checkbox"
    >
      <input
        id="Духовка"
        name="Духовка"
        type="checkbox"
      />
      <label
        for="Духовка"
      >
        Духовка
      </label>
    </div>
    <div
      class="checkbox"
    >
      <input
        id="Кофеварка"
        name="Кофеварка"
        type="checkbox"
      />
      <label
        for="Кофеварка"
      >
        Кофеварка
      </label>
    </div>
    <div
      class="checkbox"
    >
      <input
        id="Микроволновая печь "
        name="Микроволновая печь "
        type="checkbox"
      />
      <label
        for="Микроволновая печь "
      >
        Микроволновая печь 
      </label>
    </div>
    <div
      class="checkbox"
    >
      <input
        id="Посуда"
        name="Посуда"
        type="checkbox"
      />
      <label
        for="Посуда"
      >
        Посуда
      </label>
    </div>
    <div
      class="checkbox"
    >
      <input
        id="Посудомоечная машина"
        name="Посудомоечная машина"
        type="checkbox"
      />
      <label
        for="Посудомоечная машина"
      >
        Посудомоечная машина
      </label>
    </div>
  </div>
</div>
`;
